contracts/token/ERC721/ERC721.sol

```diff
@@ -354,6 +528,81 @@ contract ERC721 is Context, ERC165, IERC721 {
         emit Approval(ownerOf(tokenId), to, tokenId);
     }
 
+    /**
+     * @dev Private function to add a token to this extension's ownership-tracking data structures.
+     * @param to address representing the new owner of the given token ID
+     * @param tokenId uint256 ID of the token to be added to the tokens list of the given address
+     */
+    function _addTokenToOwnerEnumeration(address to, uint256 tokenId) private {
+        _ownedTokensIndex[tokenId] = _ownedTokens[to].length;
+        _ownedTokens[to].push(tokenId);
+    }
+
+    /**
+     * @dev Private function to add a token to this extension's token tracking data structures.
+     * @param tokenId uint256 ID of the token to be added to the tokens list
+     */
+    function _addTokenToAllTokensEnumeration(uint256 tokenId) private {
```

Comment:
All of the private functions to add/remove tokens to enumerations are a bit problematic IMO because they shouldn't be used twice on the same id without checking. They don't do these checks themselves. What do you think?

We might want to tackle this in a separate PR anyway.

Reply:
These functions are about to be removed and replaced by `EnumerableSet` and `EnumerableMap`, so I wouldn't worry about that.